src/sync/mpsc/tip.rs

```diff
@@ -0,0 +1,135 @@
+use std::sync::{Arc, Mutex};
+
+use task::{self, Task};
+use {Sink, Stream, AsyncSink, Async, Poll, StartSend};
+
+/// Tip is very similar to unbounded channel but only stores last value sent
+///
+/// I.e. if you want to send some value between from producer to a consumer
+/// and if consumer is slow it should may skip old values, the tip is
+/// a structure for the task.
+
+/// The transmission end of a channel which is used to send values
+///
+/// If the receiver is not fast enough only the last value is preserved and
+/// other ones are discarded.
+#[derive(Debug)]
+pub struct TipSender<T> {
+    inner: Arc<Mutex<Inner<T>>>,
+}
+
+/// The receiving end of a channel which preserves only the last value
+#[derive(Debug)]
+pub struct TipReceiver<T> {
+    inner: Arc<Mutex<Inner<T>>>,
+}
+
+#[derive(Debug)]
+struct Inner<T> {
+    closed: bool,
+    value: Option<T>,
+    task: Option<Task>,
+}
+
+trait AssertKindsSender: Send + Sync + Clone {}
+impl AssertKindsSender for TipSender<u32> {}
+
+trait AssertKindsReceiver: Send + Sync {}
+impl AssertKindsReceiver for TipReceiver<u32> {}
+
+impl<T> TipSender<T> {
+    /// Sets the new new value of the stream and notifies the consumer if any
+    pub fn set(&self, value: T) {
```

Comment:
This may want to return some form of `Result` to reacquire ownership of this value if the receiver has gone away

Reply:
I'd consider naming this `swap` and always return the previous value.

So the fn would be:

```
pub fn swap(&self, value: T) -> Result<Option<T>, SendError<T>> { ... }
```